Amplify/Categories/Geo/GeoCategoryBehavior.swift

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// Behavior of the Geo category that clients will use
+public protocol GeoCategoryBehavior {
+
+    // MARK: - Search
+
+    /// Search for places or points of interest.
+    /// - Parameters:
+    ///   - text: The place name or address to be used in the search. (case insensitive)
+    ///   - area: The area (.near or .boundingBox) for the search. (optional)
+    ///   - countries: Limits the search to the given a list of countries/regions. Use
+    ///   ISO-3166 3-digit country codes. (optional)
+    ///   - maxResults: The maximum number of results returned per request. (optional,
+    ///   default: 50)
+    ///   - placeIndexName: The name of the Place Index to query. (optional, default: The
+    ///   default Place Index in amplifyconfiguration.json)
+    ///   - completionHandler: The completion handler receives a Response object.  The
+    ///   success case provides a Place array.
+    func search(for text: String, // swiftlint:disable:this function_parameter_count
```

Comment:
I think this might be a little different than the auth options since according to the auth docs those are "Extra plugin specific options, only used in special circumstances", whereas some combination of these will probably be used on nearly every search.  Also, it's really just the search string, completion handler, and 4 optional parameters.  With that said, I could see where this is too specific to Amazon Location, so perhaps some of these should be broken out into an options struct that can be specified in the plugin or something? I think at the very least placeIndexName is too Amazon Location specific and was perhaps renamed by JS.  I'll check with the JS team and see how they ended up handling this.  My biggest concern with putting everything is a separate options struct is that I don't want the developer to lose simple command completion and Xcode-provided doc comment support and have to refer to separate documentation for the search options.